toolchain/check/cpp/import.cpp

```diff
@@ -2254,23 +2254,31 @@ static auto IsIncompleteClass(Context& context, SemIR::NameScopeId scope_id)
 static auto MapConstant(Context& context, SemIR::LocId loc_id,
                         clang::Expr* expr) -> SemIR::InstId {
   CARBON_CHECK(expr, "empty expression");
-  auto* integer_literal = dyn_cast<clang::IntegerLiteral>(expr);
-  if (!integer_literal) {
+
+  SemIR::TypeId type_id = MapType(context, loc_id, expr->getType()).type_id;
+  if (!type_id.has_value()) {
     context.TODO(
         loc_id, "Unsupported: constant type: " + expr->getType().getAsString());
     return SemIR::ErrorInst::InstId;
   }
-  SemIR::TypeId type_id =
-      MapType(context, loc_id, integer_literal->getType()).type_id;
-  if (!type_id.has_value()) {
-    CARBON_DIAGNOSTIC(InCppConstantMapping, Error, "invalid integer type");
-    context.emitter().Emit(loc_id, InCppConstantMapping);
-    return SemIR::ErrorInst::InstId;
+  SemIR::InstId inst_id = SemIR::InstId::None;
+  if (auto* integer_literal = dyn_cast<clang::IntegerLiteral>(expr)) {
+    auto int_id =
+        context.ints().Add(integer_literal->getValue().getSExtValue());
+    inst_id = AddInstInNoBlock<SemIR::IntValue>(
+        context, loc_id, {.type_id = type_id, .int_id = int_id});
```

Comment:
Per discussion, you want `MakeImportedLocIdAndInst` here, something like:

```suggestion
        context, MakeImportedLocIdAndInst(loc_id, {.type_id = type_id, .int_id = int_id}));
```

Reply:
In the above, note you also need to move the template parameter. :)

Reply:
Done, thanks.